clients/service/src/lib.rs

```diff
@@ -67,7 +71,28 @@ impl<Config: Clone + Send + 'static, F: Fn()> ConnectionManager<Config, F> {
 	pub async fn start<S: Service<Config, InnerError>, InnerError: fmt::Display>(
 		&self,
 	) -> Result<(), Error<InnerError>> {
+		let mut restart_delay_timer_in_secs = 20; // set default to 20 seconds for restart
+		let mut time_as_of_recording = SystemTime::now();
+
 		loop {
+			let time_now = SystemTime::now();
```

Comment:
The retry/reconnect portion, as explained on the 1st point. 
The waiting time is the variable **`restart_delay_timer_in_secs`**, which means the restart is delayed in 20 seconds.